gateway/1.5/modules/ROOT/pages/policies-pdk-debug-local.adoc

```diff
@@ -1,5 +1,145 @@
-= Debugging Custom Policies Locallys
+= Debugging Custom Policies with PDK
 ifndef::env-site,env-github[]
 include::_attributes.adoc[]
 endif::[]
-:imagesdir: ../assets/images
\ No newline at end of file
+:imagesdir: ../assets/images
+
+The Policy Development Kit (PDK) provides a preconfigured API instance located in the `<root-directory>/test/config` directory to begin testing your policy.
+
+After you successfully xref:policies-pdk-compile-policies.adoc[compile your custom policy project] and add a <<register, Flex Gateway `registration.yaml`>> file to the configuration directory, you can use the `run` command to deploy the API instance and your policy to begin testing.
+
+NOTE: PDK only supports Flex Gateways running in a Docker container.
+
+To test your policy, complete the following steps:
+
+. <<before-you-begin>>
+. <<register>>
+. <<configure-the-policy>>
+. <<add-additional-policies>> if necessary
+. <<deploy-the-policy-with-pdk>>
+
+== Before You Begin
+
+Ensure that you have:
+
+* xref:policies-pdk-compile-policies.adoc[Compiled you custom policy]
+* https://www.docker.com/[Installed Docker ^]
+
+[[register]]
+== Register a Flex Gateway Instance in Local Mode
+
+To begin using the `run` command to debug your custom policy, you must register a Flex Gateway in Local Mode.
+
+For the `run` command to work, a Flex Gateway `registration.yaml` file must be in `<root-directory>/test/config` directory. An easy way to do this is to run the registration command in the directory, or you can move a `registration.yaml` file of a previously registered Flex gateway to the directory.
+
+The `run` command only supports Flex Gateways running in Local Mode in a Docker Container. To register a Flex Gateway, see the relevant tutorial for your preferred credentail type:
+
+* xref:local-reg-run-up.adoc#docker[]
+* xref:local-reg-run-app.adoc#docker[]
+* xref:local-reg-run-token.adoc#docker[]
+
+IMPORTANT: You only need to to run the registration command. Do not run the Docker start command. The `run` command completes this step.
+
+A different registration file must be created on each device testing the policy.
+
+The registration file is ignored for storage in remote repositories and is listed in .gitignore file located at the root of the project directory.
+
+
+== Configure the Policy
+
+Depending on the configurations listed in the `gcl.yaml` file you configured in xref:policies-pdk-create-schema-definition.adoc[], you may have to configure some parameters before running the policy locally.
+
+For example, if the `definition/gcl.yaml` file is as follows:
+
+[source,yaml]
+----
+apiVersion: gateway.mulesoft.com.v1alpha1
+kind: Extension
+metadata:
+  labels:
+    title: my-custom-policy
+    category: Security 
+spec:
+  extends:
+    - name: extension-definition
+  properties: 
+    user:
+      type: string
+      default: user1
+    password:
+      type: string
+  required:
+    - someConfiguration
```

Comment:
This require value here is wrong. It references the properties above in case they are required. In this case, these are the following possible configurations:

1. Omit the `required` property (so no property is required)
2. user required
```
  required:
    - user
```
3. password required
```
  required:
    - password
```
4. user and password required
```
  required:
    - user
    - password
```

Reply:
I think, given the example you gave, we should mark the password as required.